lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -263,6 +275,12 @@ def _format_datetime(dt):
         ),
         "allow_anonymous": course.allow_anonymous,
         "allow_anonymous_to_peers": course.allow_anonymous_to_peers,
+        "user_roles": user_roles,
+        "user_is_privileged": bool(user_roles & {
+            FORUM_ROLE_ADMINISTRATOR,
+            FORUM_ROLE_MODERATOR,
+            FORUM_ROLE_COMMUNITY_TA,
```

Comment:
Can you explain this code?
```
bool(user_roles & {
            FORUM_ROLE_ADMINISTRATOR,
            FORUM_ROLE_MODERATOR,
            FORUM_ROLE_COMMUNITY_TA,
})
```

Reply:
This checks if there is an intersection of roles between the user roles and the roles listed here. 

user_roles is a set of roles that the user has, and we are using the `&` operator to take a set intersection. So if one of the user's roles is administrator, moderator or TA then we'll have a set with one or more entries. If the user has none of those roles we'll have an empty set. If we take a bool of that we get True if the user has one of these listed roles and False if the user has none.